### PR TITLE
Fix broken internal link

### PR DESCRIPTION
**Pull request by bwignall**
[Current version](https://github.com/typelevel/cats/blob/main/CONTRIBUTING.md#write-code) of the link 404s.

### CODE REVIEW

**Review by satorg — DISMISSED** [review]
Thanks!

**Comment by satorg** [comment]
`FutureSuite` on JVM is failing again: [link](https://github.com/typelevel/cats/actions/runs/9530488675/job/26270414424?pr=4622#step:13:7064)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ builds:
 
 ### Write code
 
-[See guidelines](guidelines.md).
+[See guidelines](docs/guidelines.md).
```
Comment (danicheg) [inline CONTRIBUTING.md:136]: I think it's fine to refer to the site page here.
```suggestion
[See guidelines](https://typelevel.org/cats/guidelines.html).
```

Comment (bwignall) [reply CONTRIBUTING.md]: Done, thank you.

**Review by danicheg — APPROVED** [review]

**Review by satorg — APPROVED** [review]

**Pull request merged**